blogs/2022/09/15/dev-container-features.md

```diff
@@ -0,0 +1,139 @@
+
+Features are self-contained units of installation code, container config, and/or settings and extensions designed to enable new development features in your dev container. They're installed through feature packs, which can be built to work with a wide range of base container images. As part of our work on the [open dev container specification](https://containers.dev/), we’ve made some improvements to where you can grab pre-created feature packs and how you can author and distribute your own.
+
+Let’s see what’s new and how you can get started with features from any dev container supporting tool or service (like the [VS Code Remote - Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) or [GitHub Codespaces](https://github.com/features/codespaces))!
+
+## Adding features to your dev container
+
+Dev container feature packs provide a quick way to associate dev container metadata with some install steps. You can add them to your dev containers through a simple reference.
+
+Feature packs can now be stored as [OCI Artifacts](https://github.com/opencontainers/artifacts) in any supporting container registry, which means you can reference them using the same types of identifiers you would use to reference a container image. We’ve moved some early feature packs that were in the [vscode-dev-containers](https://github.com/microsoft/vscode-dev-containers/issues/1589) repository into a new [devcontainers/features](https://github.com/devcontainers/features) repository where they are published using this new approach.
+
+Adding different features from the feature packs in the devcontainers/features repository is as simple as adding a `features` property to your `devcontainer.json`. Each feature pack has a `README.md` that shows how to reference its feature and which options are available for it.
+
+The example below installs the [go](https://github.com/devcontainers/features/tree/main/src/go) and [docker-in-docker](https://github.com/devcontainers/features/tree/main/src/docker-in-docker) features using these feature packs:
+
+```json
+"name": "my-project-devcontainer",
+"image": "mcr.microsoft.com/devcontainers/base:ubuntu",
+"features": {
+    "ghcr.io/devcontainers/features/go:1": {
+        "version": "1.18"
+    },
+    "ghcr.io/devcontainers/features/docker-in-docker:1": {
+        "version": "latest",
+        "moby": true
+    }
+}
+```
+
+You can also explore official and publicly contributed feature packs [on the spec site](https://containers.dev/features). Any feature pack can be added by editing `devcontainer.json`, and publicly published ones can be added through existing dev container configuration experiences (like what you’d see in the [VS Code Remote - Containers extension](https://code.visualstudio.com/docs/remote/containers#_dev-container-features-preview)).
+
+![Spec site list of available feature packs](./site-feature-list.png)
```

Comment:
Can you retake the image? It's low res and hard to read.

Reply:
Done! Sorry about that - it often loses quality it once had when I convert from Word to saving it as an image I can use in markdown.